compiler/AST/AstPrintDocs.cpp

```diff
@@ -98,16 +112,27 @@ bool AstPrintDocs::enterModSym(ModuleSymbol* node) {
   }
 
   // If this is a sub module (i.e. other modules were entered and not yet
-  // exited before this one), ensure the docs naming is correct.
-  if (!this->moduleNames.empty()) {
-    node->addPrefixToName(this->moduleNames.top() + ".");
+  // exited before this one), we want to open a new file in a subdirectory.
+  if (strcmp(node->name, this->moduleName.c_str()) != 0) {
```

Comment:
I'm pretty sure you could just us == here since this->moduleName is a std:string